hat/assets/js/apps/Iaso/domains/entities/components/EntitiesQuerybuilder/utils.ts

```diff
@@ -0,0 +1,162 @@
+import { JsonLogicTree } from '@react-awesome-query-builder/mui';
+import { QueryBuilderFields } from 'bluesquare-components';
+import { Form } from 'Iaso/domains/forms/types/forms';
+
+export const borderColor = 'rgba(0, 0, 0, 0.23)';
+export type LogicOperator = 'and' | 'or';
+export type NotState = boolean;
+
+export type ShortForm = {
+    id?: number;
+    form_name?: string;
+    form_id?: string;
+};
+
+export type FormState = {
+    form: ShortForm;
+    logic?: JsonLogicTree;
+    operator?: string;
+    fields?: QueryBuilderFields;
+};
+export const defaultFormState: FormState = {
+    form: {
+        id: undefined,
+        form_name: undefined,
+        form_id: undefined,
+    },
+    logic: undefined,
+    operator: undefined,
+};
+export const operatorButtons = [
+    {
+        key: 'and',
+        label: 'AND',
+        alwaysVisible: false,
+    },
+    {
+        key: 'or',
+        label: 'OR',
+        alwaysVisible: false,
+    },
+];
+export const parseInitialLogic = (logicInput: any, formsList: Form[]) => {
+    let parsedNot = false;
+    let logic = logicInput;
+    if (!logic)
+        return {
+            parsedNot,
+            mainOperator: 'and',
+            parsedFormStates: [],
+        };
+    if (typeof logic === 'string') {
+        try {
+            logic = JSON.parse(logic);
+        } catch {
+            return {
+                parsedNot,
+                mainOperator: 'and',
+                parsedFormStates: [],
+            };
+        }
+    }
+    if (logic['!']) {
+        parsedNot = true;
+        logic = logic['!'];
+    }
```

Comment:
I think you can save a few lines and avoid using `let` with reassignments:

```suggestion
    const fallback = {
        parsedNot:false,
        mainOperator: 'and',
        parsedFormStates: [],
    };
    if (!logicInput)
        return fallback;

    try {
        const logic = (typeof logicInput) === 'string' ? JSON.parse(logic) : logicInput;
    } catch {
        return fallback;
    }

    const parsedNot = Boolean( logic['!'])
    const logicToUse = logic['!'] || logic
```